http-server-netty/src/main/java/io/micronaut/http/server/netty/websocket/NettyServerWebSocketHandler.java

```diff
@@ -345,8 +356,25 @@ protected Object invokeExecutable(BoundExecutable boundExecutable, MethodExecuti
     }
 
     private Object invokeExecutable0(BoundExecutable boundExecutable, MethodExecutionHandle<?, ?> messageHandler) {
-        return ServerRequestContext.with(originatingRequest,
-                (Supplier<Object>) () -> boundExecutable.invoke(messageHandler.getTarget()));
+        return this.executorSelector.select(messageHandler.getExecutableMethod(), threadSelection)
+            .map(
+                executorService -> {
+                    ReturnType<?> returnType = messageHandler.getExecutableMethod().getReturnType();
+                    if (returnType.isReactive()) {
+                        return Mono.from((Publisher<?>) boundExecutable.invoke(messageHandler.getTarget()))
+                            .subscribeOn(Schedulers.fromExecutor(executorService))
+                            .contextWrite(reactorContext -> reactorContext.put(ServerRequestContext.KEY, originatingRequest));
+                    } else {
+                        return executorService.submit(() -> ServerRequestContext.with(originatingRequest,
```

Comment:
what happens if the method throws an exception in this code path. is it dropped or logged?

Reply:
@yawkat Good question...it appears to just get dropped right now because there's no error handling logic specific to `Future` in `AbstractNettyWebSocketHandler`, but there is for `Publisher`. I'm thinking the simplest approach may be to have the `ExectueOn` path always return a `Publisher`.